blockchain/state.go

```diff
@@ -0,0 +1,258 @@
+package blockchain
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	flowsdk "github.com/onflow/flow-go-sdk"
+
+	"github.com/dapperlabs/flow-playground-api/model"
+	"github.com/dapperlabs/flow-playground-api/storage"
+
+	"github.com/golang/groupcache/lru"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+// improvement: create instance pool as a possible optimization. We can pre-instantiate empty
+// instances of emulators waiting around to be assigned to a project if init time will be proved to be an issue
+
+// NewState creates an instance of the state with provided storage access and caching.
+func NewState(store storage.Store, cache *lru.Cache) *State {
+	return &State{
+		store: store,
+		cache: cache,
+		mu:    map[uuid.UUID]*sync.RWMutex{},
+	}
+}
+
+// State implements stateful operations on the blockchain, keeping records of transaction executions.
+//
+// State exposes API to interact with the blockchain but also makes sure the state is persisted and
+// implements state recreation with caching and resource locking.
+type State struct {
```

Comment:
`State` is a bit of a confusing name to me - at first I thought it was the underlying state layer for this module, but it's actually the main entry point.

What about calling `blockchain.Projects` (or similar) to reflect that it holds a map of active blockchain projects?

Reply:
It make sense to me honestly, but now you saying it makes me realise you might be more correct :D Will think about changing to Project or some other. Thanks